### PR TITLE
Bugfix/unr 4619 remove legacy rpcs

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
Remove legacy RPCs + fix GDK latency tests 

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -1988,7 +1988,7 @@ void USpatialNetDriver::TickFlush(float DeltaTime)
 #endif // WITH_SERVER_CODE
 	}
 
-	if (SpatialGDKSettings->UseRPCRingBuffer() && Sender != nullptr)
+	if (Sender != nullptr)
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1991]: Can assume this to be true?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp [hunk 2]
```diff
@@ -428,7 +428,7 @@ bool USpatialLatencyTracer::ContinueLatencyTrace_Internal(const AActor* Actor, c
 	}
 
 	// We do minimal internal tracking for native rpcs/properties
-	const bool bInternalTracking = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking() || Type == ETraceType::Tagged;
+	const bool bInternalTracking = Type == ETraceType::Tagged;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp:431]: Can you add a TODO here with a link back to the original ticket

File: SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h [hunk 3]
```diff
@@ -477,7 +443,6 @@ const TMap<Worker_ComponentId, FString> ServerAuthorityWellKnownComponents = {
 	{ ALWAYS_RELEVANT_COMPONENT_ID, "unreal.AlwaysRelevant" },
 	{ DORMANT_COMPONENT_ID, "unreal.Dormant" },
 	{ VISIBLE_COMPONENT_ID, "unreal.Visible" },
-	{ SERVER_RPC_ENDPOINT_COMPONENT_ID_LEGACY, "unreal.UnrealServerRPCEndpointLegacy" },
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h:480]: Delete the well known schema also, will also require updating the tests :(

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1991]: No, the net driver can tick before the Spatial connection is established (which is when Sender is created)

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 4]
```diff
@@ -187,25 +183,8 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
...
-		if (RPCsOnEntityCreation* QueuedRPCs = OutgoingOnCreateEntityRPCs.Find(Actor))
-		{
-			if (QueuedRPCs->HasRPCPayloadData())
-			{
-				ComponentDatas.Add(QueuedRPCs->CreateRPCPayloadData());
-			}
-			OutgoingOnCreateEntityRPCs.Remove(Actor);
-		}
-	}
+	checkf(RPCService != nullptr, TEXT("Attempting to send an RPC with a null RPCService."));
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:186]:
```suggestion
	checkf(RPCService != nullptr, TEXT("Attempting to create an entity with a null RPCService."));
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp (hunk 2 above)
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp:431]: Why has this changed?

**Review by improbable-valentyn — APPROVED** [review]

**Review by m-samiec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp]: This is the `fix GDK latency tests` additional that I requested as mentioned in description - one line change that didn't need to be in this PR but means we don't have to wait another 45 mins to cut the rc.

File: SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h (hunk 3 above)
Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h:480]: The file still remains! Has the server->server rpcs in it still

**Pull request merged**